### PR TITLE
Sprint 17 success stories

### PR DESCRIPTION
**Pull request by Tehsurfer**
# Description

This is a sprint 17 ticket defined here:
https://www.wrike.com/open.htm?id=678413958

A preview of this branch can be viewed here:
https://sparc-succcess-stories.herokuapp.com/news-and-events/community-spotlight

Changes:
 - Split community spotlight into fireside chats, success stories and community spotlights
 - Added the success stories page back 
 - Created a 'Fireside chats' page, which displays all available fireside chats. This is a new content model called 'Fireside Chats' on contentful
 - Created a 'Community Announcements' page, which displays news items. This pulls content from the 'news' content model on contentful
 
 
 #### Community spotlight page changes:
https://sparc-succcess-stories.herokuapp.com/news-and-events/community-spotlight
image

 
#### Fireside chats page:
https://sparc-succcess-stories.herokuapp.com/news-and-events/community-spotlight/fireside-chats
image

#### Community Announcements page:
https://sparc-succcess-stories.herokuapp.com/news-and-events/community-spotlight/community-announcements
image

## Type of change

Delete those that don't apply.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Been tested locally and on the sparc-app clone here:
https://sparc-succcess-stories.herokuapp.com/news-and-events/community-spotlight

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have utilized components from the Design System Library where applicable
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: components/CommunityAnnouncements/CommunityAnnouncementItem.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  color: rgb(131, 0, 191);
+  font-family: Asap;
+  font-size: 14px;
+  font-weight: 500;
+  border: 1px solid $median;
+  color: $median;
+  &:hover {
+    color: #1a1489;
+  }
+}
```
Comment (hsorby) [inline components/CommunityAnnouncements/CommunityAnnouncementItem.vue:132]: Is there a reason for this styling here? Shouldn't this be coming from the design components?

File: components/CommunityAnnouncements/CommunityAnnouncementListings.vue [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+@import '@/assets/_variables.scss';
+
+.seperator-path {
+  width: 100%;
+  height: 2px;
+  background: rgb(216, 216, 216);
+  border-radius: 0px;
+  margin-bottom: 32px;
+  margin-top: 32px;
+}
```
Comment (hsorby) [inline components/CommunityAnnouncements/CommunityAnnouncementListings.vue:58]: REM units not px?

File: pages/news-and-events/community-spotlight/fireside-chats/index.vue [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+<style lang="scss" scoped>
+@import '@/assets/_variables.scss';
+.page-wrap {
+  max-width: 80em;
+}
+
+.subpage {
+  margin-bottom: 20px;
+}
```
Comment (hsorby) [inline pages/news-and-events/community-spotlight/fireside-chats/index.vue:104]: REM units

File: pages/news-and-events/community-spotlight/index.vue [hunk 4]
```diff
@@ -76,26 +134,20 @@ export default {
...
 </script>
 
 <style lang="scss" scoped>
 @import '@/assets/_variables.scss';
-.page-wrap {
-  max-width: 80em;
+h2 {
+  font-size: 1.5rem;
+  line-height: 2.25rem;
+  font-size: 30px;
```
Comment (hsorby) [inline pages/news-and-events/community-spotlight/index.vue:146]: WHy does this not come from the design components?

File: pages/news-and-events/community-spotlight/success-stories/index.vue [hunk 5]
```diff
@@ -99,8 +99,4 @@ export default {
 .page-wrap {
   max-width: 80em;
 }
-
-.subpage {
-  margin-bottom: 20px;
```
Comment (hsorby) [inline pages/news-and-events/community-spotlight/success-stories/index.vue:104]: REM units

**Review by hsorby — CHANGES_REQUESTED** [review]
Just a couple of px units left to convert.  And a conflict.

**Comment by Tehsurfer** [comment]
> Just a couple of px units left to convert. And a conflict.

Ok I'm going to squash some of these commits as they aren't really necessary

File: pages/news-and-events/community-spotlight/success-stories/index.vue (hunk 5 above)
Comment (Tehsurfer) [reply pages/news-and-events/community-spotlight/success-stories/index.vue:104]: Fixed now

File: pages/news-and-events/community-spotlight/community-announcements/index.vue [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+<style lang="scss" scoped>
+@import '@/assets/_variables.scss';
+.page-wrap {
+  max-width: 80em;
+}
+
+.subpage {
+  margin-bottom: 20px;
+}
```
Comment (hsorby) [inline pages/news-and-events/community-spotlight/community-announcements/index.vue:105]: REM units

File: pages/news-and-events/community-spotlight/fireside-chats/index.vue (hunk 3 above)
Comment (hsorby) [inline pages/news-and-events/community-spotlight/fireside-chats/index.vue:104]: REM units

File: pages/news-and-events/community-spotlight/success-stories/_id.vue [hunk 7]
```diff
@@ -298,8 +307,8 @@ export default {
   height: 2px;
```
Comment (hsorby) [inline pages/news-and-events/community-spotlight/success-stories/_id.vue:307]: REM units

**Review by hsorby — COMMENTED** [review]
I have found some more uses of the px as a units.

File: components/CommunityAnnouncements/CommunityAnnouncementItem.vue (hunk 1 above)
Comment (egauzens) [reply components/CommunityAnnouncements/CommunityAnnouncementItem.vue]: Agreed, you should update all the font stylings in this PR to use the classes from the design system: https://nih-sparc.github.io/sparc-design-system-components/?path=/docs/components-typography--fonts#body-styling

So for example, the .item-description you can remove and just make the class name .body1
image

File: components/CommunityAnnouncements/CommunityAnnouncementListings.vue [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+.subpage {
+  margin-bottom: 20px;
```
Comment (egauzens) [inline components/CommunityAnnouncements/CommunityAnnouncementListings.vue]: You should be able to use a spacing class from the design system here: https://nih-sparc.github.io/sparc-design-system-components/?path=/docs/layout-spacing--spacing-values#margin

So in this case you can just use either mb-16 or mb-24

**Review by egauzens — CHANGES_REQUESTED** [review]

File: components/CommunityAnnouncements/CommunityAnnouncementListings.vue [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  cursor: pointer;
+  display: block;
+  font-size: 1rem;
+  font-weight: 700;
+  padding: 0;
+  &:hover,
+  &:active {
+    text-decoration: underline;
+  }
+}
```
Comment (hsorby) [inline components/CommunityAnnouncements/CommunityAnnouncementListings.vue:71]: Why is this link decoration not coming from the design system?

File: pages/news-and-events/community-spotlight/success-stories/_id.vue [hunk 10]
```diff
@@ -295,25 +301,22 @@ export default {
...
   font-weight: 600;
   text-transform: uppercase;
 }
 
 .story-field {
-  font-family: Asap;
   font-size: 14px;
-  line-height: 24px;
+  line-height: 1.5rem;
 }
```
Comment (hsorby) [inline pages/news-and-events/community-spotlight/success-stories/_id.vue:320]: Is there not something in the design system for this? If not there should be.

**Review by hsorby — COMMENTED** [review]
I still feel that there a few things here that should be coming from the design system.  If they don't exist in the design system they should be added.

File: components/CommunityAnnouncements/CommunityAnnouncementListings.vue (hunk 9 above)
Comment (Tehsurfer) [reply components/CommunityAnnouncements/CommunityAnnouncementListings.vue:71]: The link specified in wireframes is different than the design system

Comment (hsorby) [reply components/CommunityAnnouncements/CommunityAnnouncementListings.vue:71]: Then tell Gazelle and use what is in the design system.

File: pages/news-and-events/community-spotlight/success-stories/_id.vue [hunk 11]
```diff
@@ -295,25 +301,22 @@ export default {
...
 .seperator-path {
   width: 100%;
-  height: 2px;
+  height: 0.125rem;
   background: rgb(216, 216, 216);
   border-radius: 0px;
-  margin-top: 32px;
-  margin-bottom: 32px;
+  margin-top: 2rem;
+  margin-bottom: 2rem;
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply pages/news-and-events/community-spotlight/success-stories/_id.vue:308]: I've looked pretty extensively and haven't found any. Happy to use one if there is one though

[the remark this answers is not in the document]
Comment (hsorby) [reply pages/news-and-events/community-spotlight/success-stories/_id.vue:308]: my-32

File: pages/news-and-events/community-spotlight/success-stories/_id.vue (hunk 10 above)
Comment (Tehsurfer) [reply pages/news-and-events/community-spotlight/success-stories/_id.vue:320]: `body4` is 14px but it is the wrong colour to match the wireframes:
https://nih-sparc.github.io/sparc-design-system-components/?path=/story/components-typography--body

File: pages/news-and-events/community-spotlight/success-stories/_id.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (hsorby) [reply pages/news-and-events/community-spotlight/success-stories/_id.vue:308]: I would raise this with Gazelle.

File: pages/news-and-events/community-spotlight/success-stories/_id.vue (hunk 10 above)
Comment (hsorby) [reply pages/news-and-events/community-spotlight/success-stories/_id.vue:320]: I would use that and let Gazelle sort it out, the wireframes should be built upon the design system.

**Review by hsorby — APPROVED** [review]
Apart from the separator thing, I'm happy for the move to using the design system.

**Comment by Tehsurfer** [comment]
@egauzens would you be able to check your requested changes have gone in?

I can't merge this in until the change request is addressed.

**Review by egauzens — APPROVED** [review]

**Pull request merged**